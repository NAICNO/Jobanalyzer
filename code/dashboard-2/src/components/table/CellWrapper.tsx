import { Box, StyleProps } from '@chakra-ui/react'
import { ReactNode } from 'react'

interface CellWrapperProps {
  styleProps?: StyleProps
  children: ReactNode
}

export default function CellWrapper({styleProps, children}: CellWrapperProps) {
  return (
    <Box padding={1} {...styleProps}>
      {children}
    </Box>
  )
}
