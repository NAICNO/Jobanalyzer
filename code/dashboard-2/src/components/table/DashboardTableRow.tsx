import { flexRender, Row } from '@tanstack/react-table'
import { Td, Tr } from '@chakra-ui/react'
import { CELL_BACKGROUND_COLORS } from '../../Constants.ts'
import React from 'react'

interface DashboardTableRowProps {
  row: Row<DashboardTableItem>
  cluster: Cluster
}

const DashboardTableRow = ({row, cluster}: DashboardTableRowProps) => {

  let rowStyles : React.CSSProperties = {
    borderBottom: '1px solid',
    borderColor: 'gray.200',
  }

  if(cluster.uptime){
    const allCells = row.getAllCells()

    const cpuStatusCell = allCells.find(cell => cell.column.id === 'cpu_status')

    if(cpuStatusCell?.getValue() !=0) {
      rowStyles = {
        ...rowStyles,
        backgroundColor: CELL_BACKGROUND_COLORS.DOWN,
      }
    }
  }

  return (
    <Tr key={row.id} style={{...rowStyles}}>
      {row.getAllCells().map((cell, cellIndex, cellArray) => {
        const meta: any = cell.column.columnDef.meta
        return (
          <Td
            key={cell.id}
            isNumeric={meta?.isNumeric}
            borderRight={cellIndex !== cellArray.length - 1 ? '1px solid' : 'none'}
            borderColor="gray.200"
            padding={0}
          >
            {flexRender(cell.column.columnDef.cell, cell.getContext())}
          </Td>
        )
      })}
    </Tr>
  )
}

export default DashboardTableRow
