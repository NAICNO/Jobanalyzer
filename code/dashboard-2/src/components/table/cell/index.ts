export { default as CellWithLink } from './CellWithLink'
export { default as CellWrapper } from './CellWrapper'
export { default as GenericCell } from './GenericCell'
export { default as GpuFieldCell } from './GpuFieldCell'
export { default as WorkingFieldCell } from './WorkingFieldCell'
