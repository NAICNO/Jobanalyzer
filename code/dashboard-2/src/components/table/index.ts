export * from './DasboardTable'
export * from './DashboardTableRow'
export * from './DeadWeightTable'
export * from './JobQueryResultsTable'
export * from './TableHeader'
export * from './TableRow'
export * from './ViolatingJobTable'
export * from './ViolatingUserTable'
