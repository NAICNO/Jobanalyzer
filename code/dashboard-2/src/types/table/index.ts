export * from './DashboardTableColumnHeader.ts'
export * from './DashboardTableItem.ts'
export * from './DeadWeightTableColumnHeader.ts'
export * from './DeadWeightTableItem.ts'
export * from './JobQueryJobId.ts'
export * from './JobQueryResultsTableColumnHeader.ts'
export * from './JobQueryResultsTableItem.ts'
export * from './ViolatingJobTableColumnHeader.ts'
export * from './ViolatingJobTableItem.ts'
export * from './ViolatingUserTableColumnHeader.ts'
export * from './ViolatingUserTableItem.ts'
