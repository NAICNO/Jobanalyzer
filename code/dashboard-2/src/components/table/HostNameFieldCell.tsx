import { Text } from '@chakra-ui/react'
import CellWrapper from './CellWrapper.tsx'
import { HOSTNAMES_ALIAS } from '../../Constants.ts'

interface HostNameFieldCellProps {
  value: string;
}

export const HostNameFieldCell = ({value}: HostNameFieldCellProps) => {
  const text = HOSTNAMES_ALIAS[value] || value
  return (
    <CellWrapper styleProps={{paddingLeft:4}}>
      <Text as='b'>{text}</Text>
    </CellWrapper>
  )
}
