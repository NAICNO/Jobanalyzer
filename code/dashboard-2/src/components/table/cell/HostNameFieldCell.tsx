import { Text } from '@chakra-ui/react'
import CellWrapper from './CellWrapper.tsx'
import { HOSTNAMES_ALIAS } from '../../../Constants.ts'

interface HostNameFieldCellProps {
  value: string;
}

const HostNameFieldCell = ({value}: HostNameFieldCellProps) => {
  const text = HOSTNAMES_ALIAS[value] || value
  return (
    <CellWrapper styleProps={{paddingLeft: 4}}>
      <Text>{text}</Text>
    </CellWrapper>
  )
}

export default HostNameFieldCell
