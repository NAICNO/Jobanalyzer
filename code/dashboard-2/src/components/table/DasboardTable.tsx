import { Table, Tbody } from '@chakra-ui/react'
import { Table as TableType } from '@tanstack/react-table'
import DashboardTableRow from './DashboardTableRow.tsx'
import DashboardTableHeader from './DashboardTableHeader.tsx'

interface DashboardTableProps {
  table: TableType<DashboardTableItem>
  cluster: Cluster
}

const DashboardTable = ({table, cluster}: DashboardTableProps) => {
  return (
    <Table size="sm" border="1px solid" borderColor="gray.200">
      <DashboardTableHeader table={table}/>
      <Tbody>
        {table.getRowModel().rows.map((row) =>
          <DashboardTableRow row={row} cluster={cluster}/>
        )}
      </Tbody>
    </Table>
  )
}

export default DashboardTable
