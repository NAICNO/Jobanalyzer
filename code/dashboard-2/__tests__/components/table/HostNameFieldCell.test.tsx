import '@testing-library/jest-dom'
import { describe, it, expect } from 'vitest'
import { render, screen } from '@testing-library/react'
import { HostNameFieldCell } from '../../../src/components/table/HostNameFieldCell'

describe('HostNameFieldCell', () => {
  it('renders correctly when value is in HOSTNAMES_ALIAS', () => {
    const value = 'ml1.hpc.uio.no' // assuming 1 is a key in HOSTNAMES_ALIAS
    const {asFragment} = render(<HostNameFieldCell value={value}/>)
    expect(asFragment()).toMatchSnapshot()
    expect(screen.getByText('ML1').parentNode)
  })

  it('renders correctly when value is not in HOSTNAMES_ALIAS', () => {
    const value = '999' // assuming 999 is not a key in HOSTNAMES_ALIAS
    const {asFragment} = render(<HostNameFieldCell value={value}/>)
    expect(asFragment()).toMatchSnapshot()
  })
})
