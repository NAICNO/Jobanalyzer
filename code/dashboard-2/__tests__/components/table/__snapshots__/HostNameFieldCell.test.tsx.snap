// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HostNameFieldCell > renders correctly when value is in HOSTNAMES_ALIAS 1`] = `
<DocumentFragment>
  <div
    class="css-1phsvop"
  >
    <b
      class="chakra-text css-0"
    >
      ML1
    </b>
  </div>
</DocumentFragment>
`;

exports[`HostNameFieldCell > renders correctly when value is not in HOSTNAMES_ALIAS 1`] = `
<DocumentFragment>
  <div
    class="css-1phsvop"
  >
    <b
      class="chakra-text css-0"
    >
      999
    </b>
  </div>
</DocumentFragment>
`;
